Cannot build openrazer-driver-dkms on 7.0.8-arch1-1, >=7.1-rc4

### Affected Device / PID

Any supported device

### Operating System / Version

Archlinux

### Kernel Version

7.0.8-arch1-1 (x86_64)

### OpenRazer Version

3.12.2

### Support Checklist

- [x] My user was added to the plugdev group.
- [x] I have rebooted and the problem persists.
- [x] I looked at the [Troubleshooting](https://github.com/openrazer/openrazer/wiki/Troubleshooting) page and it did not solve the problem.

### Symptoms

- [ ] Uninstalling OpenRazer fixes the problem.
- [ ] Re-plugging the device fixes the problem.
- [x] It worked before a system/kernel update.
- [ ] Only a problem if the daemon is running.
- [ ] Only a problem from power up or a reboot.
- [ ] Dual boot with Windows (Synapse not installed).
- [ ] Dual boot with Windows (Synapse 3 installed).
- [ ] Dual boot with Windows (Synapse 4 installed).
- [ ] Intermittent or erratic problem.
- [ ] My device is missing a feature.

### Please describe the issue

With the new kernel 7.0.8-arch1-1 on Archlinux, the dkms does not compile anymore:
```
DKMS (dkms-3.4.1) make.log for openrazer-driver/3.12.2 for kernel 7.0.8-arch1-1 (x86_64)
lör 16 maj 2026 09:04:32 CEST

Building module(s)
# command: KERNELDIR=/lib/modules/7.0.8-arch1-1/build make driver

:: Compiling OpenRazer kernel modules
========================================
make -C /lib/modules/7.0.8-arch1-1/build M=/var/lib/dkms/openrazer-driver/3.12.2/build/driver modules
make[1]: Entering directory '/usr/lib/modules/7.0.8-arch1-1/build'
make[2]: Entering directory '/var/lib/dkms/openrazer-driver/3.12.2/build/driver'
  CC [M]  razerkbd_driver.o
razerkbd_driver.c: In function ‘razer_raw_event_bitfield’:
razerkbd_driver.c:4837:29: error: too few arguments to function ‘hid_report_raw_event’; expected 6, have 5
 4837 |                             hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), 0);
      |                             ^~~~~~~~~~~~~~~~~~~~
In file included from razerkbd_driver.c:11:
/usr/lib/modules/7.0.8-arch1-1/build/include/linux/hid.h:1271:5: note: declared here
 1271 | int hid_report_raw_event(struct hid_device *hid, enum hid_report_type type, u8 *data,
      |     ^~~~~~~~~~~~~~~~~~~~
razerkbd_driver.c:4841:29: error: too few arguments to function ‘hid_report_raw_event’; expected 6, have 5
 4841 |                             hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), 0);
      |                             ^~~~~~~~~~~~~~~~~~~~
/usr/lib/modules/7.0.8-arch1-1/build/include/linux/hid.h:1271:5: note: declared here
 1271 | int hid_report_raw_event(struct hid_device *hid, enum hid_report_type type, u8 *data,
      |     ^~~~~~~~~~~~~~~~~~~~
make[4]: *** [/usr/lib/modules/7.0.8-arch1-1/build/scripts/Makefile.build:289: razerkbd_driver.o] Error 1
make[3]: *** [/usr/lib/modules/7.0.8-arch1-1/build/Makefile:2108: .] Error 2
make[2]: *** [/usr/lib/modules/7.0.8-arch1-1/build/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/var/lib/dkms/openrazer-driver/3.12.2/build/driver'
make[1]: *** [Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/usr/lib/modules/7.0.8-arch1-1/build'
make: *** [Makefile:38: driver] Error 2

# exit code: 2
# elapsed time: 00:00:01
```

I am not sure if [this link (changelog for the kernel version)](https://cdn.kernel.org/pub/linux/kernel/v7.x/ChangeLog-7.0.8) will help, but reading the make.log (above) clearly points that a specific function definition got changed in the newer kernel version.

--- a/driver/razerkbd_driver.c
+++ b/driver/razerkbd_driver.c
@@ -11,6 +11,7 @@
 #include <linux/hid.h>
 #include <linux/dmi.h>
 #include <linux/input-event-codes.h>
+#include <linux/version.h>
 
 #include "usb_hid_keys.h"
 
@@ -4835,11 +4836,23 @@ static int razer_raw_event_bitfield(struct hid_device *hdev, struct razer_kbd_us
 
                             // report key down
                             xdata[1] = cur_value;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0) || \
+        (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 10) && LINUX_VERSION_CODE < KERNEL_VERSION(7, 1, 0)) || \
+        (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 18, 33) && LINUX_VERSION_CODE < KERNEL_VERSION(6, 19, 0))
+                            hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), sizeof(xdata), 0);
+#else
                             hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), 0);
+#endif
 
                             // report key up
                             xdata[1] = 0x00;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0) || \
+        (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 10) && LINUX_VERSION_CODE < KERNEL_VERSION(7, 1, 0)) || \
+        (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 18, 33) && LINUX_VERSION_CODE < KERNEL_VERSION(6, 19, 0))
+                            hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), sizeof(xdata), 0);
+#else
                             hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), 0);
+#endif
                         }
                     }
                 }

